packages/pubsub/src/connection-pool.js

```diff
@@ -459,4 +477,65 @@ ConnectionPool.prototype.shouldReconnect = function(status) {
   return true;
 };
 
+/**
+ *
+ */
+ConnectionPool.prototype.watchForReady = function() {
+  var self = this;
+
+  this.on('newListener', function(eventName) {
+    if (eventName == CHANNEL_READY && !self.waiting) {
+      clearImmediate(self.waitHandle);
```

Comment:
I don't follow the clear & then setImmediate pattern here. What case does this handle?

Reply:
I've since changed it, but this can fire off roughly 5 times at almost the same exact time, but it only should be called once.